### PR TITLE
feat: page layout for form with vehicle reference

### PR DESCRIPTION
**Pull request by talamcol**
References [DM-156](https://autoricardo.atlassian.net/browse/DM-156)

## Motivation and context

We would like to use the same layout for each checkout form that is connected to a vehicle. 

## Before

No page layout

## After

New page layout for forms

## How to test

The new component is under: Layout / Pages / Form with vehicle reference

### CODE REVIEW

File: src/components/layout/WithVehicleReference.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  title,
+  backLink,
+  vehicle,
+  children,
+}) => {
+  const Component = chakra('main');
+
+  return (
+    <Center>
+      <Component maxWidth="container.xl">
```
Comment (lkappeler) [inline src/components/layout/WithVehicleReference.tsx:29]: Should we use the container component instead? https://chakra-ui.com/docs/components/container

File: src/components/layout/WithVehicleReference.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        >
+          <GridItem area="header">
+            {backLink ? (
+              <chakra.div paddingBottom="xl">
+                <Link href={backLink.url} leftIcon={<ArrowLeftIcon />}>
+                  {backLink.text}
+                </Link>
+              </chakra.div>
+            ) : null}
+            <chakra.h1 textStyle="heading1">{title}</chakra.h1>
```
Comment (lkappeler) [inline src/components/layout/WithVehicleReference.tsx:47]: should we use the Heading component? https://chakra-ui.com/docs/components/heading

File: src/components/layout/WithVehicleReference.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  vehicle,
+  children,
+}) => {
+  const Component = chakra('main');
+
+  return (
+    <Center>
+      <Component maxWidth="container.xl">
+        <Grid
+          templateAreas={{
```
Comment (lkappeler) [inline src/components/layout/WithVehicleReference.tsx:31]: This does not match the Figma file, the header is aligned with the image and the backlink takes the whole width
Screenshot 2022-08-03 at 12 19 03

File: src/components/layout/WithVehicleReference.tsx (hunk 1 above)
Comment (lkappeler) [reply src/components/layout/WithVehicleReference.tsx:29]: Also don't we need to ensure the container takes the full 1280px?

File: src/components/layout/WithVehicleReference.tsx (hunk 3 above)
Comment (lkappeler) [reply src/components/layout/WithVehicleReference.tsx:31]: Not fully sure but I think you need to specify a 12 gird to match the design, something along this lines maybe https://codepen.io/legatore/pen/rNddBNM (estimated the gird gap)
(gen: https://grid.layoutit.com/?id=YvDcDKe)
image

**Review by lkappeler — COMMENTED** [review]
Nice start, some questions from my side, besides the alignment with the design. 
Do you think we should go more generic with the layouts / create some sort of base layout with container and page margins, a grid that matches the page as there might be cases where the right side will not be a vehicle, so I wonder whether the layout should be more generic in accepting the content and providing the sections and spacings according to the design? Lemme know if you wanna have a chat 🙂

File: src/components/layout/WithVehicleReference.tsx (hunk 1 above)
Comment (Averethel) [reply src/components/layout/WithVehicleReference.tsx:29]: Wouldn't `<Container as="main">` do the trick?

File: src/components/layout/WithVehicleReference.tsx (hunk 3 above)
Comment (talamcol) [reply src/components/layout/WithVehicleReference.tsx:46]: done

**Review by lkappeler — COMMENTED** [review]
I think the car section is in the wrong spot on mobile, otherwise, its looking fine

image

**Comment by talamcol** [comment]
> I think the car section is in the wrong spot on mobile, otherwise, its looking fine
> 
> image

thanks. Messed it up during refactoring

File: src/components/layout/WithVehicleReference.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        url: 'https://www.autoscout24.ch/de',
+      },
+      vehicle: {
+        vehicleTitle: 'AUDI A5 Sportback 3.0 TDI quattro S-tronic (Limousine)',
+        price: `CHF 23'900.–`,
+        sellerName: `Auto-Center Grenchen AG 2540 Grenchen (SO)`,
+        sellerAddress: `2540 Grenchen (SO)`,
+        image: <img src="/dummyImage.jpeg" alt="vehicle reference image" />,
+      },
+      children: (
```
Comment (Averethel) [inline src/components/layout/WithVehicleReference.stories.mdx:34]: Can you exclude the children from the args table? You'd need to add `argTypes` prop with:
```
{
   children: {
      table: {
        disable: true,
      },
    },
}
```

File: src/components/layout/WithVehicleReference.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        <Stack spacing="md">
+          <Input name="name" placeholder="Name" />
+          <Input name="email" placeholder="E-Mail address" />
+          <Input name="phone" placeholder="Phone number" />
+        </Stack>
+      ),
```
Comment (Averethel) [inline src/components/layout/WithVehicleReference.stories.mdx:40]: You can add `args` and `argTypes` on `Meta`, this way you won't need to repeat them in every story, but just overwrite the ones that you need.

File: src/components/layout/WithVehicleReference.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      }}
+      gridTemplateRows="minmax(min-content, max-content) minmax(min-content, max-content) 1fr"
+    >
+      <GridItem area="backlink">
+        {backLink ? (
+          <Link href={backLink.url} leftIcon={<ArrowLeftIcon />}>
+            {backLink.text}
+          </Link>
+        ) : null}
+      </GridItem>
```
Comment (Averethel) [inline src/components/layout/WithVehicleReference.tsx:43]: Can we ditch the grid item if there is no backlink:
```suggestion
      {backLink ? (
        <GridItem area="backlink">
          <Link href={backLink.url} leftIcon={<ArrowLeftIcon />}>
            {backLink.text}
          </Link>
        <GridItem area="backlink">
      ) : null}
```
?

File: src/components/vehicleReference/index.tsx [hunk 7]
```diff
@@ -52,3 +52,4 @@ const VehicleReference: FC<Props> = ({
 };
 
 export default VehicleReference;
+export { Props as VehicleProps };
```
Comment (Averethel) [inline src/components/vehicleReference/index.tsx:55]:
```suggestion
export { Props as VehicleReferenceProps };
```
(to keep consistency with other prop exports)

**Review by Averethel — APPROVED** [review]
Really nice, thank you for simplifying 🙏🏻

**Review by lkappeler — APPROVED** [review]
Nice effort 👍

**Pull request merged**